src/corona_hakab_model/population_analysis/population_analysis.py

```diff
@@ -0,0 +1,81 @@
+import math
+import pickle
+
+import numpy as np
+from generation.circles_generator import PopulationData
+from generation.connection_types import ConnectionTypes
+from generation.matrix_generator import MatrixData
+from matplotlib import pyplot as plt
+
+
+class PopulationAnylzer:
+    """
+    this module is in charge of plotting statistics about the generation stage, and specifically the population
+    """
+
+    __slots__ = "population_data", "matrix_data"
+
+    def __init__(self, population_data_path, matrix_data_path=None):
+
+        # importing population data from path
+        try:
+            with open(population_data_path, "rb") as population_data:
+                self.population_data: PopulationData = pickle.load(population_data)
+        except FileNotFoundError:
+            raise FileNotFoundError("population analyzer couldn't open population data")
+
+        # todo i saw in the issue that population analysis should accept a matrix data.
+        # todo why is it necessary? it should only analyze population...
+        # importing matrix data from path.
+        # for now, allows not importing a matrix data
+        if matrix_data_path is not None:
+            try:
+                with open(matrix_data_path, "rb") as matrix_data:
+                    self.matrix_data: MatrixData = pickle.load(matrix_data)
+            except FileNotFoundError:
+                raise FileNotFoundError("population analyzer couldn't open matrix data")
+
+    def plot_circles_sizes(self):
+        """
+        plotting a bar chart for each connection type, showing the circle sizes count.
+        """
+        fig = plt.figure()
+        types_amount = len(ConnectionTypes)
+        grid_count = math.ceil(math.sqrt(types_amount))
+        for i, con_type in enumerate(ConnectionTypes, 1):
+            ax = fig.add_subplot(grid_count, grid_count, i)
+            ax.set_title(str(con_type))
+            ax.set_ylabel("amount of circles")
+            ax.set_xlabel("circle size")
+            circles = self.population_data.social_circles_by_connection_type[con_type]
+            size_count = {}
+            for circle in circles:
+                size = circle.agent_count
+                if size not in size_count:
+                    size_count[size] = 0
+                size_count[size] += 1
+            # creating a bar graph with a bar for each circle size
+            bars = [size for size in sorted(size_count.keys())]
+            heights = [size_count[size] for size in bars]
```

Comment:
A suggestion - can retrieve both bars and heights like so:
bars, heights = zip(*sorted(size_count.items()))